### ISSUE TITLE
[regression] "You seem to be initializing a member variable with itself"

### ISSUE THREAD

**Issue by lgalfaso** [cpplint/cpplint#461]
The code below used to be understood without reporting an error, now it is reporting "You seem to be initializing a member variable with itself"

```c++
class foo {
  foo(bool bar): bar(bar) {}

 private:
  bool bar;
};
```

**Comment by yangfan-yf-yf** [comment]
I couldn't reproduce `runtime/init` with the example as posted.

I ran the exact snippet against both cpplint 2.0.2 and current `develop` at `96db7ae`:

```console
python cpplint.py --filter=-,+runtime/init issue461.cpp
Done processing issue461.cpp
```

Both runs exited with status 0. With the default filters, both versions report only `legal/copyright` and `runtime/explicit`, not `runtime/init`.

The `runtime/init` matcher is unchanged between these revisions. It only matches an underscore-suffixed member initialized from itself, such as `bar_(bar_)` or `bar_(CHECK_NOTNULL(bar_))`; `bar(bar)` does not match it.

Could you provide the following so I can reproduce the reported regression?

- the output of `cpplint --version`;
- the exact command and any relevant `CPPLINT.cfg`;
- the complete diagnostic line, including its category;
- the exact source line that receives the diagnostic, plus adjacent lines if it differs from the reduced example.

If this came through an editor or wrapper, please also include the cpplint executable and version it invokes.
